### PR TITLE
gentoo: stop passing USE environment to emerge

### PR DESCRIPTION
**Pull request by 257**
also make sure we stay with systemd-9999 (version built from git) in order keep up with changes in systemd

### CODE REVIEW

File: .github/mkosi.pkgmngr/gentoo/etc/portage/binrepos.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[binhost]
+sync-uri = https://raw.githubusercontent.com/257/binpkgs/master
```
Comment (DaanDeMeyer) [inline .github/mkosi.pkgmngr/gentoo/etc/portage/binrepos.conf:2]: Why bother? We already set `PORTAGE_BINHOST` anyway

Comment (257) [reply .github/mkosi.pkgmngr/gentoo/etc/portage/binrepos.conf]: dropped, never mind.

File: .github/mkosi.pkgmngr/gentoo/etc/portage/package.use/systemd [hunk 2]
```diff
@@ -0,0 +1 @@
+sys-apps/systemd boot
```
Comment (DaanDeMeyer) [inline .github/mkosi.pkgmngr/gentoo/etc/portage/package.use/systemd:1]: Can you get rid of the `boot` flag upstream when systemd v254 releases? Given that gnu-efi isn't necessary anymore from v254 onwards I'm not sure if putting sd-boot behind a flag still provides much value

Comment (257) [reply .github/mkosi.pkgmngr/gentoo/etc/portage/package.use/systemd:1]: i'm not the maintainer but i see it's used in number of places [here](https://github.com/gentoo/gentoo/blob/master/sys-apps/systemd/systemd-9999.ebuild):
```
...
boot? ( $(python_gen_any_dep 'dev-python/pyelftools[${PYTHON_USEDEP}]') )
...
$(meson_native_use_bool boot bootloader)
...
```

Comment (DaanDeMeyer) [reply .github/mkosi.pkgmngr/gentoo/etc/portage/package.use/systemd:1]: Yeah unless gentoo really cares about the extra python build dependencies, I would get rid of `boot` in the systemd package and just build the boot stuff by default.

**Comment by 257** [comment]
@DaanDeMeyer can you verify if the fail is related to this commit?

**Comment by DaanDeMeyer** [comment]
> @DaanDeMeyer can you verify if the fail is related to this commit?

How old is the systemd we're running? The failure we're seeing should be fixed in git main of upstream.

**Comment by 257** [comment]
got HEAD

**Comment by DaanDeMeyer** [comment]
> git HEAD

And we're rebuilding that every time and not using a binary package or so that might be older?

**Comment by 257** [comment]
yes, good catch.

**Comment by behrmann** [comment]
CI failure for gentoo disk looks unrelated and is only when powering down the machine. Merging.

**Pull request merged**